conductor_api/src/interface_impls/http.rs

```diff
@@ -13,12 +15,21 @@ impl HttpInterface {
 }
 
 impl Interface for HttpInterface {
-    fn run(&self, handler: IoHandler, kill_switch: Receiver<()>) -> Result<(), String> {
+    fn run(
+        &self,
+        handler: IoHandler,
+        kill_switch: Receiver<()>,
+    ) -> Result<(Broadcaster, thread::JoinHandle<()>), String> {
         let url = format!("0.0.0.0:{}", self.port);
-        let _server = ServerBuilder::new(handler)
+
+        let server = ServerBuilder::new(handler)
             .start_http(&url.parse().expect("Invalid URL!"))
             .map_err(|e| e.to_string())?;
-        let _ = kill_switch.recv();
-        Ok(())
+        let broadcaster = Broadcaster::Noop;
```

Comment:
So HTTP implements a broadcaster but it's really just a no-op. So you've also set up a pattern that's extendable into different interfaces being implemented...

Reply:
This sounds right.  @maackle it seems like this same method could be used to implement a `LongPollHttpInterface` that could do broadcasting over http, it just waits to "send" the messages when it gets a poll http request.

Reply:
Exactly. Each new Interface has the opportunity to also specify a Broadcaster, or use Noop if it doesn't support two-way messaging.

Reply:
You could use it to do HTTP/2 server-side events too...